modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala

```diff
@@ -115,4 +156,30 @@ object ConfigurableSuite {
         localDateTime <- localDateTimeArbitrary.arbitrary
         zoneId <- zoneIdArbitrary.arbitrary
       } yield localDateTime.toDateTime(zoneId))
+
+  implicit val instantArbitrary: Arbitrary[Instant] =
+    Arbitrary(Arbitrary.arbitrary[Long].map(new Instant(_)))
+
+  implicit val intervalArbitrary: Arbitrary[Interval] =
+    Arbitrary(Arbitrary.arbitrary[(DateTime, DateTime)]
+      .filter(i => i._1.isBefore(i._2))
+      .map { i =>
+        // Comparing new Interval(foo) with Interval.parseWithOffset(foo) will return false
+        // See http://www.joda.org/joda-time/apidocs/org/joda/time/Interval.html#parseWithOffset-java.lang.String-
+        Interval.parseWithOffset(new Interval(i._1, i._2).toString)
+      })
+
+  implicit val durationArbitrary: Arbitrary[Duration] =
+    Arbitrary(Arbitrary.arbitrary[Long].map(new Duration(_)))
+
+  implicit val periodArbitrary: Arbitrary[Period] = {
+    Arbitrary(
+      for {
+        // Marshelling and unmarshalling fails with big values
```

Comment:
Did you try generating a `Long` and constructing a `Period` using the single-argument constructor? I expect that may give you more consistent results with broader coverage.